### ISSUE TITLE
[Feature Request] CNN layer slicing

### ISSUE THREAD

**Issue by alnah005** [tenstorrent/tt-metal#26137]
### Is your feature request related to a problem? Please describe

Currently, we are working on enabling the following models in high resolution (1280x800 for some and 2000x2000 for others):

```
    yolov4
    yolov7
    yolov8x
    yolov8s
    yolov8s_world
    yolov11
    yolov12n
    resnet50
    mobilenetv2
    efficientnetb0
    efficientnetb3
    efficientnetb4
    vgg_unet
    ufld_v2
    segformer_classification
    swin_transformer
    swin_transformer_v2
```

### Describe the solution you'd like

The models above can run (there are existing demos) on low resolution. Ideally, only turning on Dram slicing for the ops that don't fit is what's required. However, only convolution is supported (#17493). The models have the following pytorch layers:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">

</head>

<body link=blue vlink=purple>

Row Labels | efficientnetb0 | efficientnetb3 | efficientnetb4 | mobilenetv2 | resnet50 | segformer_classification | swin_transformer | swin_transformer_v2 | ufld_v2 | vgg_unet | yolov11 | yolov12n | yolov4 | yolov7 | yolov8s | yolov8s_world | yolov8x | Grand   Total
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
torch.ops.aten.view | 1 | 1 | 1 | 1 | 1 | 189 | 635 | 609 | 5 |   | 11 | 57 |   | 6 | 5 | 64 | 5 | 1591
torch.ops.aten.convolution | 81 | 130 | 160 | 52 | 53 | 18 | 1 | 1 | 37 | 25 | 88 | 120 | 110 | 95 | 64 | 71 | 104 | 1210
torch.ops.aten.native_batch_norm | 49 | 78 | 96 | 52 | 53 |   |   |   | 36 | 8 | 81 | 113 | 107 | 92 | 57 | 61 | 97 | 980
torch.ops.aten.silu_ | 49 | 78 | 96 |   |   |   |   |   |   |   | 77 | 81 |   |   | 57 | 57 | 97 | 592
torch.ops.aten.permute |   |   |   |   |   | 49 | 144 | 144 |   |   | 1 | 40 |   | 3 |   | 33 |   | 414
torch.ops.aten.clone |   |   |   |   |   | 10 | 168 | 166 |   |   | 1 | 36 |   | 3 |   | 1 |   | 385
torch.ops.aten._unsafe_view |   |   |   |   |   | 24 | 123 | 147 |   |   | 3 | 40 |   |   |   | 1 |   | 338
torch.ops.aten.add.Tensor |   |   |   | 10 |   | 16 | 84 | 94 |   |   | 16 | 36 | 23 | 9 | 8 | 16 | 20 | 332
torch.ops.aten.addmm | 1 | 1 | 1 | 1 | 1 | 41 | 145 | 121 | 2 |   |   |   |   |   |   | 5 |   | 319
torch.ops.aten.expand |   |   |   |   |   | 32 | 96 | 144 |   |   | 4 | 32 |   |   |   | 3 |   | 311
torch.ops.aten.slice.Tensor |   |   |   |   |   |   | 144 | 114 | 8 |   | 6 |   |   | 12 |   |   |   | 284
torch.ops.aten.mul.Tensor | 16 | 26 | 32 |   |   |   |   | 24 |   |   | 2 | 9 | 72 | 12 | 1 | 13 | 1 | 208
torch.ops.aten.cat |   |   |   |   |   |   | 3 | 3 |   | 4 | 23 | 22 | 10 | 18 | 19 | 20 | 19 | 141
torch.ops.aten.bmm |   |   |   |   |   | 16 | 48 | 48 |   |   | 2 | 16 |   |   |   | 9 |   | 139
torch.ops.aten.native_layer_norm |   |   |   |   |   | 30 | 53 | 53 |   |   |   |   |   |   |   | 2 |   | 138
torch.ops.aten.transpose.int |   |   |   |   |   | 28 | 26 | 26 |   |   | 2 | 25 |   |   | 1 | 2 | 1 | 111
torch.ops.aten.relu_ |   |   |   |   | 49 |   |   |   | 33 | 24 |   |   |   |   |   |   |   | 106
torch.ops.aten.add_.Tensor | 9 | 19 | 25 |   | 16 | 8 |   |   | 16 |   |   |   |   |   |   |   |   | 93
torch.ops.aten.div.Tensor |   |   |   |   |   | 8 | 24 | 48 |   |   | 1 | 1 |   |   | 1 | 9 | 1 | 93
torch.ops.aten.silu |   |   |   |   |   |   |   |   |   |   |   |   |   | 89 |   |   |   | 89
torch.ops.aten.sigmoid | 16 | 26 | 32 |   |   |   |   |   |   | 1 | 1 | 1 |   | 3 | 1 | 5 | 1 | 87
torch.ops.aten.mean.dim | 17 | 27 | 33 | 1 | 1 | 1 | 1 | 1 |   |   |   |   |   |   |   |   |   | 82
torch.ops.aten.softplus |   |   |   |   |   |   |   |   |   |   |   |   | 72 |   |   |   |   | 72
torch.ops.aten.tanh |   |   |   |   |   |   |   |   |   |   |   |   | 72 |   |   |   |   | 72
torch.ops.aten._softmax |   |   |   |   |   | 8 | 24 | 24 |   |   | 2 | 9 |   |   | 1 | 2 | 1 | 71
torch.ops.aten.gelu |   |   |   |   |   | 8 | 24 | 24 |   |   |   |   |   |   |   |   |   | 56
torch.ops.aten.clamp_min |   |   |   |   |   |   |   | 48 |   |   |   |   |   |   |   | 3 |   | 51
torch.ops.aten.linalg_vector_norm |   |   |   |   |   |   |   | 48 |   |   |   |   |   |   |   | 3 |   | 51
torch.ops.aten.constant_pad_nd |   |   |   |   |   |   | 24 | 24 |   |   |   |   |   |   |   |   |   | 48
torch.ops.aten.roll |   |   |   |   |   |   | 24 | 22 |   |   |   |   |   |   |   |   |   | 46
torch.ops.aten.split.Tensor |   |   |   |   |   |   |   |   |   |   | 8 | 4 |   |   | 9 | 9 | 9 | 39
torch.ops.aten.mm |   |   |   |   |   | 8 | 3 | 27 |   |   |   |   |   |   |   |   |   | 38
torch.ops.aten.leaky_relu_ |   |   |   |   |   |   |   |   |   |   |   |   | 35 |   |   |   |   | 35
torch.ops.aten.hardtanh_ |   |   |   | 35 |   |   |   |   |   |   |   |   |   |   |   |   |   | 35
torch.ops.aten.max_pool2d_with_indices |   |   |   |   | 1 |   |   |   | 1 | 4 | 3 |   | 3 | 8 | 3 | 3 | 3 | 29
torch.ops.aten.unsqueeze |   |   |   |   |   |   | 1 | 1 |   |   | 8 |   |   |   |   | 18 |   | 28
torch.ones | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 17
torch.ops.aten.split_with_sizes |   |   |   |   |   |   |   |   |   |   | 3 | 9 |   |   | 1 | 1 | 1 | 15
torch.ops.aten.upsample_nearest2d |   |   |   |   |   |   |   |   |   |   | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 14
torch.ops.aten.sub.Tensor |   |   |   |   |   |   |   |   |   |   | 2 | 2 |   | 3 | 2 | 2 | 2 | 13
torch.ops.aten.unbind.int |   |   |   |   |   |   |   |   |   |   | 8 |   |   |   |   |   |   | 8
torch.ops.aten.copy_ |   |   |   |   |   |   |   |   |   |   |   |   |   | 6 |   |   |   | 6
torch.ops.aten.max.dim |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   | 4 |   | 4
torch.ops.aten.convolution_transpose |   |   |   |   |   |   |   |   |   | 4 |   |   |   |   |   |   |   | 4
torch.ops.aten.pow.Tensor_Scalar |   |   |   |   |   |   |   |   |   |   |   |   |   | 3 |   |   |   | 3
torch.ops.aten.adaptive_max_pool2d |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   | 3 |   | 3
torch.ops.aten.squeeze.dim |   |   |   |   |   |   | 1 | 1 |   |   |   |   |   |   |   |   |   | 2
torch.ops.aten.as_strided_ |   |   |   |   |   |   | 1 | 1 |   |   |   |   |   |   |   |   |   | 2
torch.ops.aten.relu |   |   |   |   |   |   |   |   | 1 |   |   |   |   |   |   |   |   | 1
(blank) |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
Grand Total | 240 | 387 | 477 | 153 | 176 | 495 | 1798 | 1964 | 140 | 71 | 356 | 656 | 507 | 365 | 233 | 423 | 365 | 8806

</body>

</html>

Not all the layers require dram slicing, but the following **pytorch layers** with the following shapes have been confirmed to require it:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">

</head>

<body link=blue vlink=purple>

Function   Call Name | Attrs | Input shapes | Output shapes
-- | -- | -- | --
max_pool2d_with_indices | 'kernel': [2, 2], 'stride': [2, 2], 'padding': [0, 0],   'dilation': [1, 1] | [1, 64, 1280, 800] | [1, 64, 640, 400]
max_pool2d_with_indices | 'kernel': [2, 2], 'stride': [2, 2], 'padding': [0, 0],   'dilation': [1, 1] | [1, 128, 640, 400] | [1, 128, 320, 200]
max_pool2d_with_indices | 'kernel': [2, 2], 'stride': [2, 2], 'padding': [0, 0],   'dilation': [1, 1] | [1, 256, 320, 200] | [1, 256, 160, 100]
max_pool2d_with_indices | 'kernel': [2, 2], 'stride': [2, 2], 'padding': [0, 0],   'dilation': [1, 1] | [1, 512, 160, 100] | [1, 512, 80, 50]
concatenate | [1, 512, 160, 100], [1, 512, 160, 100] | [1, 1024, 160, 100]
concatenate | [1, 256, 320, 200], [1, 256, 320, 200] | [1, 512, 320, 200]
concatenate | [1, 128, 640, 400], [1, 128, 640, 400] | [1, 256, 640, 400]
concatenate | [1, 64, 1280, 800], [1, 64, 1280, 800] | [1, 128, 1280, 800]

</body>

</html>

> Note: the requirement table above is not comprehensive and as we investigate the full model layers above, we will potentially add more layers to the requirement table.

[all_cnn_models.xlsx](https://github.com/user-attachments/files/21552187/all_cnn_models.xlsx)

### Describe alternatives you've considered

Sharding, Tile/Row major

**Comment by pavlejosipovic** [comment]
max_pool2d_with_indices we don't have working yet even w/o slicing?

**Comment by pavlejosipovic** [comment]
This is a customer team feature request, pls do not close
